### PR TITLE
[GPU] Use int64_t type for axis in softmax

### PR DESCRIPTION
**Pull request by Lyamin-Roman**
### Details:
 - *Enum changed to int64_t for softmax primitive.*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/softmax.cpp [hunk 1]
```diff
@@ -31,6 +31,11 @@ std::string softmax_inst::to_string(softmax_node const& node) {
 
     std::stringstream primitive_description;
 
+    json_composite softmax_info;
+    softmax_info.add("input id", node.input().id());
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/softmax.cpp:35]: I think it's not needed here. We should either use "dependencies" field to track inputs in the dump or add dependencies ids dumping into the common code for all primitives.

File: src/plugins/intel_gpu/src/graph/impls/ocl/softmax.cpp [hunk 2]
```diff
@@ -13,6 +13,28 @@
 namespace cldnn {
 namespace ocl {
 
+static inline kernel_selector::softmax_dim GetSoftmaxDim(int64_t axis, size_t rank) {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/impls/ocl/softmax.cpp:16]: nit: get_softmax_dim

File: src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp [hunk 3]
```diff
@@ -25,24 +25,13 @@ namespace cldnn {
...
-
     /// @brief Constructs softmax primitive.
     /// @param id This primitive id.
     /// @param input Input primitive id.
-    /// @param dimension Defines a scope of normalization (see #dimension).
+    /// @param dimension Defines a scope of normalization
     softmax(const primitive_id& id,
             const primitive_id& input,
-            const dimension_t dimension = normalize_fyx,
+            const int64_t dimension = 1,
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp:34]: let's remove default value for dimension

File: src/plugins/intel_gpu/src/plugin/ops/softmax.cpp [hunk 4]
```diff
@@ -14,34 +14,33 @@
...
+static void CreateSoftmaxOp(Program& p, const std::shared_ptr<ngraph::op::v8::Softmax>& op) {
     p.ValidateInputs(op, {1});
     auto inputPrimitives = p.GetInputPrimitiveIDs(op);
     std::string layerName = layer_type_name_ID(op);
+
+    int64_t axis = op->get_axis();
+    size_t rank = op->get_input_shape(0).size();
+    if (axis < 0)
+        axis += rank;
+    if (axis < 0 || axis >= static_cast<int64_t>(rank))
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/ops/softmax.cpp:38]: Why it's needed? The axis is supposed to be normalized later in the plugin

File: src/plugins/intel_gpu/tests/test_cases/reorder_gpu_test.cpp [hunk 5]
```diff
@@ -1602,7 +1602,7 @@ TEST(reorder_gpu_opt, mean_mul_val)
     cldnn::mem_lock<float> ptr(output, get_test_stream());
     float* a_ptr = answers;
     for (auto& val : ptr)
-        EXPECT_FLOAT_EQ(*(a_ptr++), val);;
+        EXPECT_FLOAT_EQ(*(a_ptr++), val);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/tests/test_cases/reorder_gpu_test.cpp:1605]: [random spot] Please add instances for Softmax-8 functional tests

File: src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp [hunk 6]
```diff
@@ -51,13 +40,12 @@ struct softmax : public primitive_base<softmax> {
...
-    /// - when set to @link softmax::dimension_t softmax::normalize_y @endlink each input column is normalized independently,
-    /// - when set to @link softmax::dimension_t softmax::normalize_z @endlink each input z-coordinate is normalized independently,
-    /// - when set to @link softmax::dimension_t softmax::normalize_f @endlink each in-depth vector of input is normalized independently,
-    /// - when set to @link softmax::dimension_t softmax::normalize_fyx @endlink each 3d image within input is normalized independently,
-    /// - when set to @link softmax::dimension_t softmax::normalize_all @endlink everything is normalized,
-    dimension_t dimension;
+    /// - when softmax dimension is set to 0 (b dim) each batch vector of input is normalized independently,
+    /// - when softmax dimension is set to 1 (f dim) each in-depth vector of input is normalized independently,
+    /// - when softmax dimension is set to 2 (y dim) each input column is normalized independently,
+    /// - when softmax dimension is set to 3 (x dim) each input row is normalized independently.
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp:46]: The description seems to be invalid for 5d case (dim=2 means Z, not Y). So you can either fix it or just remove description at all as there's no any plugin specific for this parameter and we can just refer to the operation spec.

File: src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp (hunk 3 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp:34]: Default value matches operation spec

File: src/plugins/intel_gpu/tests/test_cases/reorder_gpu_test.cpp (hunk 5 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/tests/test_cases/reorder_gpu_test.cpp:1605]: Softmax-8 functional tests already exist

File: src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp (hunk 6 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/softmax.hpp:46]: Yes, this is a special case for an example. The phrase "Being given a 4-dimensional input.." speaks of this.

File: src/plugins/intel_gpu/src/plugin/ops/softmax.cpp (hunk 4 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/plugin/ops/softmax.cpp:38]: The axis is normalized in the same place as in other operations (for example: gather_elements, cum_sum, interpolate)

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**